algorithms/src/sorting/impl/Kokkos_SortByKeyImpl.hpp

```diff
@@ -177,17 +177,37 @@ void sort_by_key_via_sort(
 
   static_assert(sizeof...(MaybeComparator) <= 1);
   if constexpr (sizeof...(MaybeComparator) == 0) {
+#ifdef KOKKOS_ENABLE_SYCL
+    auto* raw_keys_in_comparator = keys_in_comparator.data();
+    auto stride                  = keys_in_comparator.stride(0);
```

Comment:
@masterleinad I'm curious, why is it 0 here?

Reply:
`extent(0)` described the distance between elements that have the same indices except for the first dimension and consecutive indices in the first dimension. Since we are only allowing one-dimensional Views here, that is the distance between consecutive elements in memory.